### PR TITLE
Introduce `Credentials` protocol implementations to hold WPCOM and WPOrg credentials. 

### PR DESCRIPTION
**Pull request by selanthiraiyan**
Part of: #8432

## Description
- Introduce `Credentials` protocol.
- Rename existing `Credentials` into `WPCOMCredentials` to hold WordPress.com auth token.
- Make `WPCOMCredentials` default credentials in `SessionManagerProtocol`.
- Add `WPOrgCredentials` to hold .org site credentials.

## Testing instructions
- Install the app above the existing version and test the app.
- Do a fresh install, login and test the app.

## Screenshots
NA

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: Yosemite/Yosemite/Base/StoresManager.swift [hunk 1]
```diff
@@ -20,7 +20,7 @@ public protocol StoresManager {
     /// Switches the internal state to Authenticated.
     ///
     @discardableResult
-    func authenticate(credentials: Credentials) -> StoresManager
+    func authenticate(credentials: WPCOMCredentials) -> StoresManager
```
Comment (itsmeichigo) [inline Yosemite/Yosemite/Base/StoresManager.swift]: Can we keep `Credentials` as the input type here? We'll need to support authenticating with WPORG credentials here too.

File: Yosemite/Yosemite/Base/SessionManagerProtocol.swift [hunk 2]
```diff
@@ -37,7 +37,7 @@ public protocol SessionManagerProtocol {
 
     /// Default Credentials.
     ///
-    var defaultCredentials: Credentials? { get set}
+    var defaultCredentials: WPCOMCredentials? { get set}
```
Comment (itsmeichigo) [inline Yosemite/Yosemite/Base/SessionManagerProtocol.swift]: I think we want to keep `Credentials` as the type here too to support WPOrg credentials later.

File: WooCommerce/Classes/Yosemite/AuthenticatedState.swift [hunk 3]
```diff
@@ -22,7 +22,7 @@ class AuthenticatedState: StoresManagerState {
 
     /// Designated Initializer
     ///
-    init(credentials: Credentials) {
+    init(credentials: WPCOMCredentials) {
```
Comment (itsmeichigo) [inline WooCommerce/Classes/Yosemite/AuthenticatedState.swift]: We should also keep `Credentials` as the input type here too.

**Comment by selanthiraiyan** [comment]
Thanks for the review and inputs, Huong!

I created a new PR with the enum approach [here](https://github.com/woocommerce/woocommerce-ios/pull/8446)

**Pull request closed without merging**